app/models/species_list.rb

```diff
@@ -201,13 +219,30 @@ def name_included(name)
 
   # After defining a location, update any lists using old "where" name.
   def self.define_a_location(location, old_name)
-    old_name = connection.quote(old_name)
-    new_name = connection.quote(location.name)
-    connection.update(%(
-      UPDATE species_lists
-      SET `where` = #{new_name}, location_id = #{location.id}
-      WHERE `where` = #{old_name}
-    ))
+    update_manager = arel_update_defined_location(location, old_name)
+    # puts(update_manager.to_sql)
+    SpeciesList.connection.update(update_manager.to_sql)
+  end
+
+  # Nimmo Note: This can be done in AR with update_all, something like:
+  #   SpeciesList.where(where: old_name).update_all(
+  #     where: new_name, location_id: location.id)
+  # but I'm not sure that is preferable because it skips validations?
```

Comment:
I think skipping validations is a bonus!

We keep coming up against skipping validations in this work you're doing, Nimmo.  I never really formulated my position on it in so many words, but I guess where I was intuitively coming down on it is: validations are there to protect us from external data, not for double-checking internal code (that's what the unit tests are for).  And since I imagine that validations are relatively expensive -- presumably at very least they must instantiate instances of every record which is changed -- they are to be avoided except where necessary.  In effect: why not run the "validations" once -- i.e., in unit tests -- instead of every single time the code is run -- AR validation callbacks?

So, in this case, I would definitely recommend using SPL.update_all over Arel.

Reply:
Thanks for the feedback on validation! This is what i was hoping to hear. It certainly sounds right, but I am too ignorant about the codebase to have a real opinion. 

From now on, i'll do the AR where possible. I would also propose that we add that gem that allows for a Rails 5.2 version of AR insert_all, when needed in a later PR.